node/src/chain/fsm.rs

```diff
@@ -679,15 +771,16 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> InSyncImpl<DB, VM, N> {
             // If the accepted block is the one requested to presync peer,
             // switch to OutOfSync/Syncing mode
             if let Some(metadata) = &metadata {
-                if let Some(presync) = &mut self.presync {
-                    if metadata.src_addr == presync.peer_addr
-                        && remote_height == presync.start_height() + 1
-                    {
-                        let res =
-                            (presync.target_blk.clone(), presync.peer_addr);
-                        self.presync = None;
-                        return Ok(Some(res));
-                    }
+                let same = self
+                    .presync
+                    .as_ref()
+                    .map(|presync| {
+                        metadata.src_addr == presync.peer_addr
+                            && remote_height == presync.start_height() + 1
```

Comment:
I think this condition is unnecesary and can lead to the presync failing in case we accept start_height+1 in the meantime (for instance due to the blocks request at startup).
Receiving a valid tip+1 should be sufficient to prove the peer has a continuation of our chain.

Reply:
I'm not completely sure about this, let's discuss it and tackle it in a follow up PR